docs/HazardAnalysis/HazardAnalysis.tex

```diff
@@ -113,6 +113,54 @@ \section{Safety and Security Requirements}
+    \item[] \emph{Associated Hazards:} HA1, HA2
+
+    \item The system shall limit the number of login attempts within a period of time before login is disabled for a cooldown period.
+    \item[] \emph{Rationale:} Malicious actors may gain access to an account by a brute-force attack. This requirement significantly reduces this possibility.
+    \item[] \emph{Associated Hazards:} HA2
+
+    \item The system shall use natural language processing to reduce the error generated by OCR reading of the receipt.
+    \item[] \emph{Rationale:} OCR can sometimes return an inaccurate reading of its input. Natural language processing can reduce this by interpreting the reading using its model rather than its literal reading.
+    \item[] \emph{Associated Hazards:} HB1
+
+    \item The system shall prompt the user to retake a photo if the receipt reading is unclear.
+    \item[] \emph{Rationale:} OCR will return incorrect data to the rest of the system if the photo is unclear. Thus requesting the user to retake the photo can mitigate this issue.
+    \item[] \emph{Associated Hazards:} HB1
+
+    \item The system shall allow the user to manually input grocery items.
+    \item[] \emph{Rationale:} Should the OCR and language model not read or recognize an item properly, the user can eliminate this error state by manually reviewing and inputting their items.
+    \item[] \emph{Associated Hazards:} HB1
+
+    \item The system shall prompt the user to grant system access to the device camera and the photo gallery.
+    \item[] \emph{Rationale:} If the application does not have camera access, it cannot take a photo. The prompt will notify the user that they must grant access before the application can take or import a photo.
+    \item[] \emph{Associated Hazards:} HB2
+
+    \item The system shall perform data validation on both front-end when the user inputs and back-end.
+    \item[] \emph{Rationale:} If data from the front-end is following strict consistent data-types from conception then all that data flowing from front to back-end is easier to verify and stay consistent.
+    \item[] \emph{Associated Hazards:} HC1
+
+    \item The system shall perform database backups on a regular basis.
+    \item[] \emph{Rationale:} If the database has a failure and loses data or is offline, the system will be unable to perform some of its functions. Backing up the database will ensure swift recovery and limited data loss.
+    \item[] \emph{Associated Hazards:} HC1, HD1, HE1
+
+    \item The system shall prompt the user to connect to the internet if the system is not connected to the internet.
+    \item[] \emph{Rationale:} The system will not function correctly without internet connection. Prompting the user to obtain internet connection will notify that user what the problem is and how they might address it.
+    \item[] \emph{Associated Hazards:} HC2, HF1
+
+    \item The system shall validate that the information is returned to the user.
```

Comment:
I don't think this wording for the title and rationale quite match the intention I was thinking. The intention was to validate the information written to the database. I looked it up and we should be able to add checks such that after a write operation, we can verify that the data is written correctly. This is to preserve correctness of the database.

The main rationale this is considered is because there's the potential for something to go wrong during the write process outside of the user's control (i.e., if we want to write "Dog" to the database, we would verify that "Dog" can be found in the database before moving on). We want to ensure correctness of the user's input so we want to make sure that nothing is going wrong on the database side that could result in incorrect financial decisions.

Reply:
is this referring to SSR10 or SSR8

Reply:
For `The system shall validate that the information is returned to the user.`